ui/src/components/Tasks/DataCollection.jsx

```diff
@@ -280,13 +280,15 @@ class DataCollection extends React.Component {
                 <InputField propName="kappa" type="number" label="Kappa" />
                 <InputField propName="kappa_phi" type="number" label="Phi" />
               </FieldsRow>
-              <FieldsRow>
-                <SelectField
-                  propName="detector_mode"
-                  label="Detector mode"
-                  list={['0', 'C18', 'C12', 'C2']}
-                />
-              </FieldsRow>
+              {this.props.initialValues?.detector_modes?.length !== 0 ? (
+                <FieldsRow>
+                  <SelectField
+                    propName="detector_mode"
```

Comment:
should this be `detector_roi_mode` then or is there also a `detector_mode`  ? :)

Reply:
I see, but the actual value used and passed to the collection routine will be contained in `detector_mode`, It will be one of the values in `detector_modes` right ?

I'm being quite picky but I often wind the trailing `s` to be quite confusing. Its often easier to add `_list` or something. 

As I understand this, `detector_modes` is the list of ROI's available and the value passed is `detector_mode` ?

Reply:
We are currently not using `detector_roi_mode` but I think it might be interesting for the future. My question was also that I can't see any mapping between `detector_mode` and `detector_mode_roi`. I think you can simply call it `detector_mode_roi` and `detector_mode_list`. I think detector_mode_roi will get passed down to collect as it already exists.